src/openrct2/rct2/S6Exporter.cpp

```diff
@@ -278,20 +296,18 @@ void S6Exporter::Export()
     uint32_t researchedTrackPiecesA[128] = {};
     uint32_t researchedTrackPiecesB[128] = {};
 
-    for (int32_t i = 0; i < RCT2_OBJECT_ENTRY_COUNT; i++)
-    {
-        const rct_object_entry* entry = get_loaded_object_entry(i);
-        void* entryData = get_loaded_object_chunk(i);
-        // RCT2 uses (void *)-1 to mark NULL. Make sure it's written in a vanilla-compatible way.
-        if (entry == nullptr || entryData == nullptr || entryData == reinterpret_cast<void*>(-1))
-        {
-            std::memset(&_s6.objects[i], 0xFF, sizeof(rct_object_entry));
-        }
-        else
-        {
-            _s6.objects[i] = *entry;
-        }
-    }
+    auto& objectMgr = OpenRCT2::GetContext()->GetObjectManager();
+    ExportObjectList(objectMgr, _s6.RideObjects, ObjectType::Ride, RCT12_MAX_RIDE_OBJECTS);
+    ExportObjectList(objectMgr, _s6.SceneryObjects, ObjectType::SmallScenery, RCT2_MAX_SMALL_SCENERY_OBJECTS);
+    ExportObjectList(objectMgr, _s6.LargeSceneryObjects, ObjectType::LargeScenery, RCT2_MAX_LARGE_SCENERY_OBJECTS);
+    ExportObjectList(objectMgr, _s6.WallSceneryObjects, ObjectType::Walls, RCT2_MAX_WALL_SCENERY_OBJECTS);
+    ExportObjectList(objectMgr, _s6.BannerObjects, ObjectType::Banners, RCT2_MAX_BANNER_OBJECTS);
+    ExportObjectList(objectMgr, _s6.PathObjects, ObjectType::Paths, RCT2_MAX_PATH_OBJECTS);
+    ExportObjectList(objectMgr, _s6.PathAdditionObjects, ObjectType::PathBits, RCT2_MAX_PATH_ADDITION_OBJECTS);
+    ExportObjectList(objectMgr, _s6.SceneryGroupObjects, ObjectType::SceneryGroup, RCT2_MAX_SCENERY_GROUP_OBJECTS);
+    ExportObjectList(objectMgr, _s6.ParkEntranceObjects, ObjectType::ParkEntrance, RCT2_MAX_PARK_ENTRANCE_OBJECTS);
+    ExportObjectList(objectMgr, _s6.WaterObjects, ObjectType::Water, RCT2_MAX_WATER_OBJECTS);
+    ExportObjectList(objectMgr, _s6.ScenarioTextObjects, ObjectType::ScenarioText, RCT2_MAX_SCENARIO_TEXT_OBJECTS);
```

Comment:
Do parks exported in this manner still load correctly in vanilla? I imagine vanilla treats it as once list, and might stop after the first empty entry it encounters.

Reply:
I don't think this is the game as all objects are mapped towards {type, index}

Reply:
Vanilla simply expects a list of 721 items (in total), with any empty slots marked by setting the object header to -1.

Reply:
Well it uses an array that has indices of the entry by a specific object type which we don't really need if we just separate them simplifying a lot of code, I have it almost done in a local branch of mine.

Reply:
To be clear, its still exported/imported as expected, I can't break that for obvious reasons. So gonna mark this resolved.